bundled/.clang-tidy

```diff
@@ -0,0 +1,6 @@
+# Disable all checks because we don't want to fix things inside
+# bundled/. Clang-tidy fails if nothing is chosen, so we just enable a single
+# harmless warning:
+
+Checks: "-*,misc-sizeof-container"
```

Comment:
that is weird, can you provide an example? It works on the tester:
https://jenkins.tjhei.info/blue/organizations/jenkins/dealii-tidy/detail/PR-7651/9/pipeline#step-36-log-800

Reply:
Running
```
 ../contrib/utilities/run_clang_tidy.sh .. &> errors
```
with `DEAL_II_FORCE_BUNDLED_BOOST=ON` gives me a 256MB file which looks like
```
[...]
/mnt/data/darndt/Sources/dealii-clang-6.0.0/build/../bundled/boost-1.62.0/include/boost/variant/visitor_ptr.hpp:40:5: error: use 'using' instead of 'typedef' [modernize-use-using,-warnings-as-errors]
/mnt/data/darndt/Sources/dealii-clang-6.0.0/build/../bundled/boost-1.62.0/include/boost/variant/visitor_ptr.hpp:46:5: error: use 'using' instead of 'typedef' [modernize-use-using,-warnings-as-errors]
/mnt/data/darndt/Sources/dealii-clang-6.0.0/build/../bundled/boost-1.62.0/include/boost/variant/visitor_ptr.hpp:50:5: error: use 'using' instead of 'typedef' [modernize-use-using,-warnings-as-errors]
[...]
```

Reply:
could this be because of the .. in the path? Can you try putting your build directory outside the source directory?

Reply:
Indeed, `..` is the problem. It works if I do
```
../contrib/utilities/run_clang_tidy.sh /mnt/data/darndt/Sources/dealii-clang-6.0.0/ &> errors
```
instead. This is strange.

Reply:
Ah, you already say that it must be an absolute patch. My bad then. I am still happy to write a workaround such that it also accepts relative paths.

Reply:
The following patch works for me
```
diff --git a/contrib/utilities/run_clang_tidy.sh b/contrib/utilities/run_clang_tidy.sh
index 7755712..4eb7630 100755
--- a/contrib/utilities/run_clang_tidy.sh
+++ b/contrib/utilities/run_clang_tidy.sh
@@ -21,7 +21,7 @@
 # Usage:
 # /contrib/utilities/run_clang_tidy.sh SRC_DIR OPTIONAL_CMAKE_ARGS
 #   with:
-#     SRC_DIR is an absolute path to a deal.II source directory
+#     SRC_DIR is an absolute or relative path to a deal.II source directory
 #     OPTIONAL_CMAKE_ARGS are optional arguments to pass to CMake
 #   make sure to run this script in an empty build directory
 #
@@ -32,6 +32,11 @@
 # grab first argument:
 SRC=$1
 shift
+# convert relative to absolute path if necessary
+case $SRC in
+  /*) ;;
+   *) SRC="$PWD/$SRC" ;;
+esac
 
 if test ! -d "$SRC/source" -o ! -d "$SRC/include" -o ! -d "$SRC/examples" -o ! -f "$SRC/CMakeLists.txt" ; then
     echo "Usage:"
```

Reply:
really? This looks like it will still make a path like "/bla/blub/../source". How about the last commit I just pushed?

Reply:
Your last commit works for me as well, but ShellCheck is complaining:
```
 ^-- SC2006: Use $(...) notation instead of legacy backticked `...`.
     ^-- SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
```